### ISSUE TITLE
Consider integrating cleanlab - cleans noisy labels (enhancement)

### ISSUE THREAD

**Issue by willsmithorg** [autogluon/autogluon#1487]
The [Cleanlab](https://github.com/cleanlab/cleanlab) project aims to clean up noisy labels.
From what I understand so far it prunes those examples it is confident are labelled erroneously.  More docs [here](https://cleanlab.readthedocs.io/en/latest/cleanlab.html).

This might help improve SOTA.

For those who like academic papers : [https://arxiv.org/abs/1911.00068](https://arxiv.org/abs/1911.00068)

Two things to note:
- the lead author is CGNorthcutt who is already aware of AutoGluon and offering help on a separate issue, [here](https://github.com/awslabs/autogluon/issues/846),
- the license is "GNU Affero General Public License" - I'm not sure if this presents a problem.

**Comment by mglowacki100** [comment]
I think it could be solved by https://github.com/autogluon/autogluon/issues/1493

**Comment by mglowacki100** [comment]
Hi @willsmithorg, @Innixma 
I've implemented draft scikit-learn wrapper for autogluon that works with cleanlab, at least with wine toy-dataset:
```
class AutogluonClassifier(BaseEstimator, ClassifierMixin):
    def __init__(self, eval_metric=None, time_limit=60, hyperparameters=None, verbosity=2):
        self.eval_metric = eval_metric
        self.time_limit = time_limit
        self.hyperparameters = hyperparameters
        self.verbosity = verbosity
        self.tabular_predictor = None
        self.classes_ = None
        self.n_features_in_ = None
        self.n_classes_ = None
        self.feature_names_in_ = None

    def fit(self, X, y):
        self.n_features_in_ = X.shape[1]
        self.feature_names_in_ = X.columns.tolist()

        X = pd.DataFrame(X)
        X['target'] = y
        self.tabular_predictor = TabularPredictor(label='target', eval_metric=self.eval_metric)
        self.tabular_predictor.fit(train_data=X, time_limit=self.time_limit, hyperparameters=self.hyperparameters, verbosity=self.verbosity)

        self.classes_ = self.tabular_predictor._learner.label_cleaner.ordered_class_labels
        self.n_classes_ = len(self.classes_)

    def predict(self, X):
        if self.tabular_predictor is None:
            raise NotFittedError("Model has not been fitted. Call 'fit' before 'predict'.")
        return self.tabular_predictor.predict(X)

    def predict_proba(self, X):
        if self.tabular_predictor is None:
            raise NotFittedError("Model has not been fitted. Call 'fit' before 'predict_proba'.")
        return self.tabular_predictor.predict_proba(X)

```
cleanlab usage:
```
label_issues_info = CleanLearning(clf=AutogluonClassifier()).find_label_issues(X_train, y_train)
```
Please let me know, if this works for your usecase and be aware that I've checked it only with wine dataset.
Full code is here:
https://colab.research.google.com/drive/1tMVNbYhu9sLsUWdq1tPabid7-eb8ZPo3?usp=sharing

**Comment by shchur** [comment]
Closing this issue assuming that the sklearn-compatible API (#1493) fixed it. Please feel free to re-open if it's not the case.
